var/spack/repos/builtin/packages/harfbuzz/package.py

```diff
@@ -150,4 +150,29 @@ def configure_args(self):
         args.extend(self.with_or_without("graphite2"))
         args.extend(self.with_or_without("coretext"))
 
+        ldflags = []
+        libs = []
+        for lib in ["bzip2", "zlib", "libpng"]:
+            spec = self.spec[lib]
+            if spec.satisfies("+shared") or spec.satisfies("libs=shared"):
+                continue
+            ldflags.append(spec.libs.ld_flags)
+            libs.append(spec.libs.link_flags)
+        cairo = self.spec["cairo"]
```

Comment:
Seeing this logic for cairo if `~shared` repeatedly. Wondering if there is a way to bake this into the cairo package (`setup_dependent_build_environment`, `libs`) instead?

This is probably a more general question that each package should set up the dependent build environment correctly, whether it's built statically or not. But maybe that only sounds good in theory but doesn't work in practice?

Reply:
Yeah I know what you mean and it's something I've been thinking about for a while. I think I mentioned in the Spack survey a couple months back that it would be nice for there to be cleaner integration of lib/linker flags when using static builds (which creates the special need for all this `if ~shared` logic). But I'm not sure what a solution there would even look like, because each build system doesn't necessarily, for instance, use `LIBS` and `LDFLAGS` variables/arguments in the same way. For the moment all I can say is that I think this approach is broadly consistent with how things have been done in any number of other packages. In the future, if nothing else, maybe some of the logic can be simplified by having a package spec's `libs` attribute provide lib/linker flags for its dependencies and not just its own libraries based on `find_library`.